detekt-gradle-plugin/src/functionalTest/kotlin/io/gitlab/arturbosch/detekt/DetektReportMergeSpec.kt

```diff
@@ -72,45 +75,47 @@ internal class DetektReportMergeSpec : Spek({
             |  }
             |}
             |""".trimMargin()
-        }
 
-        it("using Groovy and Kotlin") {
-            listOf(
-                groovy to groovyBuildFileContent,
-                kotlin to kotlinBuildFileContent
-            ).forEach { (builder, mainBuildFileContent) ->
-                val projectLayout = ProjectLayout(numberOfSourceFilesInRootPerSourceDir = 0).apply {
-                    addSubmodule(
-                        name = "child1",
-                        numberOfSourceFilesPerSourceDir = 2,
-                        numberOfCodeSmells = 2
-                    )
-                    addSubmodule(
-                        name = "child2",
-                        numberOfSourceFilesPerSourceDir = 4,
-                        numberOfCodeSmells = 4
-                    )
-                }
+        fun scenarios(): List<Arguments> = listOf(
+            arguments(groovy, groovyBuildFileContent),
+            arguments(kotlin, kotlinBuildFileContent)
+        )
+
+        @ParameterizedTest(name = "Using {0}")
+        @MethodSource("scenarios")
```

Comment:
Don't commit this! For sure it will not compile.
```suggestion
        class Scenarios : ArgumentsProvider {
           override fun provideArguments(context: ExtensionContext) = Stream.of(
            arguments(DslTestBuilder.groovy(), groovyBuildFileContent),
            arguments(DslTestBuilder.kotlin(), kotlinBuildFileContent),
         )
        }

        @ParameterizedTest(name = "Using {0}")
        @ArgumentsSource(DetektReportMergeSpec.Scenarios::class)
```

But what do you think of something like this? It avoid `String` references that are easier to break. The problem is that it's more verbose. It's just a suggestion.

Reply:
I think the string reference is OK in this case, given the function sits directly above the test method. Also I just tried and IntelliJ will update the string in the annotation if the method is renamed using Refactor -> Rename in the IDE, and also if the string itself is renamed with Refactor -> Rename.